integration_tests/benchmarks/layers/layers_benchmarks.ts

```diff
@@ -127,24 +129,30 @@ function getRandomInputsAndOutputs(model: tfl.LayersModel, batchSize: number):
     }
 
     let ys: tfc.Tensor|tfc.Tensor[] = [];
-    for (const output of model.outputs) {
-      ys.push(tfc.randomUniform([batchSize].concat(output.shape.slice(1))));
-    }
-    if (ys.length === 1) {
-      ys = ys[0];
+    if (model instanceof tfl.LayersModel) {
+      for (const output of model.outputs) {
+        ys.push(tfc.randomUniform([batchSize].concat(output.shape.slice(1))));
+      }
+      if (ys.length === 1) {
+        ys = ys[0];
+      }
     }
 
     return {xs, ys};
   });
 }
 
 /** Call await data() on tensor(s). */
-async function syncData(tensors: tfc.Tensor|tfc.Tensor[]) {
-  if (Array.isArray(tensors)) {
+async function syncData(tensors: tfc.Tensor|tfc.Tensor[]|NamedTensorMap) {
+  if (tensors instanceof tfc.Tensor) {
+    await (tensors as tfc.Tensor).data();
+  } else  if (Array.isArray(tensors)) {
     const promises = tensors.map(tensor => tensor.data());
     await Promise.all(promises);
   } else {
-    await tensors.data();
+    // tensors is a NamedTensorMap.
+    const promises = Object.entries(tensors).map(item => item[1].data());
```

Comment:
should wait for data of all items for each entry.

Reply:
This is already doing that. `item` is a [key, value] pair. The key is ignored, while the value is `item[1]` here.